muon/_core/preproc.py

```diff
@@ -756,6 +764,18 @@ def func(x):
             data.raw._n_obs = data.raw.X.shape[0]
 
     else:
+        # Subset .obs
+        data._obs = data.obs[obs_subset]
+        data._n_obs = data.obs.shape[0]
```

Comment:
Not sure this is necessary for `AnnData` - also please consider de-dupping all this logic - you can use `getattr(data, f"{key}m")` and make `key: Literal["obs", "var"]` an argument to a helper